donuts/image.py

```diff
@@ -0,0 +1,380 @@
+from astropy import log
+from astropy import units as u
+from scipy import (
+    ndimage,
+    conjugate,
+    polyfit,
+)
+from scipy.fftpack import fft, ifft
+
+
+class Image(object):
+    '''Low level class which handles the image transformations and cross
+    correlation with another ``Image`` class.
+    '''
+
+    def __init__(self, data, header=None):
+        self.raw_image = data
+        self.header = header or {}
+
+        self.raw_region = None
+        self.sky_background = None
+        self.backsub_region = None
+        self.exposure_time_value = None
+        self.proj_x = None
+        self.proj_y = None
+        self.x = None
+        self.y = None
+
+    def normalise(self, exposure_keyword='EXPOSURE'):
+        '''Convert the image data into ADU per second
+
+        Parameters
+        ----------
+        exposure_keyword : str
+            Fits header keyword for exposure time. The default is `EXPTIME`.
+
+        Returns
+        -------
+        self : ``Image``
+            The current ``Image`` instance
+
+        Raises
+        ------
+        RuntimeError if the image region has not been trimmed (see the ``trim``
+        method).
+        '''
+
+        try:
+            self.exposure_time_value = self.header[exposure_keyword]
+        except KeyError:
+            log.warning(
+                'Exposure time keyword "{0}" not found, assuming 1.0'.format(
+                    exposure_keyword)
+            )
+            self.exposure_time_value = 1.0
+
+        if self.raw_region is None:
+            raise RuntimeError('Image region has not been computed.'
+                               'Please ensure the `#trim` method has been called')
+
+        self.raw_region = self.raw_region / self.exposure_time_value
+        return self
+
+    def trim(self, prescan_width=0, overscan_width=0, border=64):
+        '''Remove the optional prescan and overscan from the image, as well
+        as the outer `n` rows/colums of the image. Finally ensure the imaging
+        region is the correct dimensions for `scipy.ndimage.resize (i.e. a
+        multiple of 16.
+
+        Parameters
+        ----------
+        prescan_width : int
+            Remove the first ``prescan_width`` columns from the image, assuming
+            the are not in the imaging region.
+
+        overscan_width : int
+            Remove the last ``overscan_width`` columns from the image, assuming
+            the are not in the imaging region.
+
+        border : int
+            Ignore the first/last ``border`` rows/columns from the image,
+            assuming that they are not "typical", a common case with edge
+            effects in CCDs.
+
+        Returns
+        -------
+        self : ``Image``
+            The current ``Image`` instance
+
+        Raises
+        ------
+        None
+
+
+        '''
+        if overscan_width > 0 and prescan_width > 0:
+            image_section = self.raw_image[:, prescan_width:-overscan_width]
```

Comment:
It turns out it is not that simple. By removing the checks, we get `self.raw_image[:, 0:-0]` which with a `2048x2048` image actually removes the last axis and gives and array of size `2048x0`. I've left this previous version in for the time being.